### PR TITLE
xds: pretty print xDS updates and service config

### LINKED ISSUE #4186: xds: Add support for a pretty printing utility library
In current xDS code, we simply use `%v` or `%+v` formatting directive when logging structs which are deeply nested.

Here is an example:
```
INFO: 2021/02/04 02:38:13 [xds] [xds-client 0xc0004b7040] ADS response received: version_info:"1612257263156605616"  resources:{[type.googleapis.com/envoy.config.cluster.v3.Cluster]:{name:"cloud-internal-istio:cloud_mp_913549178901_8613762901959030423"  type:EDS  eds_cluster_config:{eds_config:{ads:{}  resource_api_version:V3}}  connect_timeout:{seconds:30}  circuit_breakers:{thresholds:{max_connections:{value:2147483647}  max_pending_requests:{value:2147483647}  max_requests:{value:2147483647}  max_retries:{value:2147483647}}}  http2_protocol_options:{max_concurrent_streams:{value:100}}  common_lb_config:{healthy_panic_threshold:{value:1}  locality_weighted_lb_config:{}}  transport_socket:{name:"envoy.transport_sockets.tls"  typed_config:{[type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext]:{common_tls_context:{tls_certificate_sds_secret_configs:{name:"tls_sds"  sds_config:{path:"/etc/envoy/tls_certificate_context_sds_secret.yaml"}}  tls_certificate_certificate_provider_instance:{instance_name:"google_cloud_private_spiffe"  certificate_name:"DEFAULT"}  combined_validation_context:{default_validation_context:{match_subject_alt_names:{exact:"spiffe://easwars-grpc-gke.svc.id.goog/ns/default/sa/example-grpc-server"}}  validation_context_sds_secret_config:{name:"validation_context_sds"  sds_config:{path:"/etc/envoy/validation_context_sds_secret.yaml"}}  validation_context_certificate_provider_instance:{instance_name:"google_cloud_private_spiffe"  certificate_name:"ROOTCA"}}  alpn_protocols:"managed-mtls"  alpn_protocols:"h2"}}}}  metadata:{filter_metadata:{key:"com.google.trafficdirector"  value:{fields:{key:"backend_service_name"  value:{string_value:"grpc-gke-helloworld-service"}}}}}  lrs_server:{self:{}}}}  type_url:"type.googleapis.com/envoy.config.cluster.v3.Cluster"  nonce:"1"
```

This is close to unreadable. We should have a simple pretty printing library which prints structs in JSON like formatting which makes it easier to read them.

### PR DESCRIPTION
**Pull request by menghanl**
fixes #4186

### CODE REVIEW

**Comment by menghanl** [comment]
This branch includes this PR, and an e2e example to show the new logs: https://github.com/menghanl/grpc-go/tree/xds_pretty_print_show

File: xds/internal/balancer/lrs/balancer.go [hunk 1]
```diff
@@ -80,6 +81,7 @@ type lrsBalancer struct {
 }
 
 func (b *lrsBalancer) UpdateClientConnState(s balancer.ClientConnState) error {
+	b.logger.Infof("Receive update from resolver, balancer config: %+v", pretty.ToJSON(s.BalancerConfig))
```
Comment (easwars) [inline xds/internal/balancer/lrs/balancer.go:84]: s/Receive/Received/

File: xds/internal/balancer/priority/balancer.go [hunk 2]
```diff
@@ -108,6 +109,7 @@ type priorityBalancer struct {
 }
 
 func (b *priorityBalancer) UpdateClientConnState(s balancer.ClientConnState) error {
+	b.logger.Infof("Receive update from resolver, balancer config: %+v", pretty.ToJSON(s.BalancerConfig))
```
Comment (easwars) [inline xds/internal/balancer/priority/balancer.go:112]: s/Receive/Received/

File: xds/internal/balancer/weightedtarget/weightedtarget.go [hunk 3]
```diff
@@ -83,6 +84,7 @@ type weightedTargetBalancer struct {
 // creates/deletes sub-balancers and sends them update. Addresses are split into
 // groups based on hierarchy path.
 func (w *weightedTargetBalancer) UpdateClientConnState(s balancer.ClientConnState) error {
+	w.logger.Infof("Receive update from resolver, balancer config: %+v", pretty.ToJSON(s.BalancerConfig))
```
Comment (easwars) [inline xds/internal/balancer/weightedtarget/weightedtarget.go:87]: s/Receive/Received/

**Review by easwars — APPROVED** [review]

**Pull request merged**